components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/AuthenticationResultCache.java

```diff
@@ -93,14 +95,33 @@ public void addToCache(AuthenticationResultCacheKey key, AuthenticationResultCac
      * @return Cached entry.
      */
     public AuthenticationResultCacheEntry getValueFromCache(AuthenticationResultCacheKey key) {
+
         AuthenticationResultCacheEntry entry = super.getValueFromCache(key);
         if (entry == null && isTemporarySessionDataPersistEnabled) {
             entry = (AuthenticationResultCacheEntry) SessionDataStore.getInstance().
                     getSessionData(key.getResultId(), CACHE_NAME);
+            if (entry != null && isCacheEntryExpired(entry)) {
+                return null;
+            }
         }
         return entry;
     }
 
+    private boolean isCacheEntryExpired(AuthenticationResultCacheEntry entry) {
+
+        String createdTimestamp = entry.getResult().getProperty(FrameworkConstants.CREATED_TIMESTAMP).toString();
+        if (createdTimestamp != null) {
+            if (FrameworkUtils.getCurrentStandardNano() >
+                    entry.getValidityPeriod() + Long.parseLong(createdTimestamp) * 1000000) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Authentication result cache is expired");
+                }
+                return true;
+            }
+        }
+        return false;
```

Comment:
Should this return false if there is no created time stamp ? Will there be scenarios where `CREATED_TIMESTAMP` is not properly set which results in the cache to not expiry ?

Reply:
If we set true here, there may be scenarios where cache entries will be discarded if `CREATED_TIMESTAMP` is not available. This is a behavioural change and may cause regressions.